### DISCUSSION TITLE
Typescript error：Type 'number' is not assignable to type 'never'

### DISCUSSION THREAD

**Discussion by ghost** [community/community#39384] [Projects and Issues] [answered]
### Select Topic Area

Bug

### Body

interface Opts {
  onFrame: () => void;
  onAudioSample: null;
  emulateSound: boolean;
  sampleRate: number;
}

class NES {
    constructor(opts: Opts) {
        this.opts = {
            onFrame() { },
            onAudioSample: null,
            emulateSound: true,
            sampleRate: 44100,
        }

        if (typeof opts !== "undefined") {
            let key: keyof Opts
            for (key in this.opts) {
                if (typeof opts[key] !== "undefined") {
                    // got err here
                    this.opts[key] = opts[key];
                }
            }
        }
    }
    opts: Opts
}

typescript: v3.8.3 error: Type 'number' is not assignable to type never. I don't understand why it is a never type.

**Comment by codpen** [comment, marked as answer]
const totalCounterColors = characters.reduce((acc: Record<string, number>, {eye_color: color}) => {
  acc[color] = (acc[color] || 0) + 1;
  return acc;
}, {});

const totalUniqueColors = Array.from(new Set(characters.map(c => c.eye_color)));

**Reply by ghost** [reply]
Perfect!
Thank you.

**Comment by felixsun55** [comment]
const totalCounterColors = characters.reduce((acc: Record<string, number>, curVal) => {
  const color = curVal.eye_color;
  if (acc[color]) {
    acc[color]++;
  } else {
    acc[color] = 1;
  }
  return acc;
}, {});

const totalUniqueColors = characters.reduce((acc: string[], curVal) => {
  if (acc.indexOf(curVal.eye_color) === -1) {
    acc.push(curVal.eye_color);
  }
  return acc;
}, []);
